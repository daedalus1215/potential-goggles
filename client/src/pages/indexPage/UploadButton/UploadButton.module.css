.uploadButton,
.uploadButton:hover {
  background-color: #3e80ed;
  color: white;
  width: 150px;
  /* margin: 4px 5px; */
  border: none;
  text-decoration: none;
  text-align: center;
  border-radius: 4px;
  font-size: .9rem;
}

.uploadButton:hover {
  background-color: #184693;
}

.wrap {
  display: flex;
}

.title {
  margin-left: 1rem;
}

.icon {
  font-size: 1.2rem;
}

.file {
  width: 60px;
  height: 40px;
  padding: 0;
  z-index: 10;
  opacity: 0;
  position: absolute;
  left: 1px;
  top: 1px;
}
