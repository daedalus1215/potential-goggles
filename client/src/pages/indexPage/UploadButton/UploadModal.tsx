import React, { useMemo, useState } from 'react';
import Modal from '@/components/modal/Modal';
import TaskDropZone from './DropZone/TaskDropZone';
import { ButtonWrapper } from '@/components';
import TagDropZone from './DropZone/TagDropZone';

type props = {
    isShowing: boolean;
    setIsShowing: (isShowing: boolean) => void;
}

const UploadModal: React.FC<props> = ({ isShowing, setIsShowing }) => {
    const [radioChoice, setRadioChoice] = useState('Tag');
    const options = ['Task', 'Tag'];

    return isShowing
        ? (<Modal setIsShowing={setIsShowing} >
            <div>
                {options.map((option) => (
                    <label key={option}>
                        <input
                            type="radio"
                            value={option}
                            checked={radioChoice === option}
                            onChange={() => setRadioChoice((option as any))}
                        />
                        {option}
                    </label>
                ))}
            </div>
            {radioChoice === 'Task'
                ? <TaskDropZone />
                : <TagDropZone />}
        </Modal>)
        : <></>;
};

export default UploadModal;