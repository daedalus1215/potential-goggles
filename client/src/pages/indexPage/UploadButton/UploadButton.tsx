import React, { useState } from 'react';
import IconButton from '../../../components/iconButton/IconButton';
import UploadModal from './UploadModal';

const UploadButton = () => {
    const [isModalShowing, setIsModalShowing] = useState(false);

    return (<>
        <IconButton
            onClick={setIsModalShowing}
            icon="bi bi-cloud-upload"
            title="Upload"
        />
        <UploadModal
            isShowing={isModalShowing}
            setIsShowing={setIsModalShowing}
            data-testid="uploadModal" />
    </>);
};

export default UploadButton;    