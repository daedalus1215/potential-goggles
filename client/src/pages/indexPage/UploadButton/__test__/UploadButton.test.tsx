import * as React from 'react';
import { render } from '@testing-library/react';
import UploadButton from '../UploadButton';

describe('src/pages/tasks/TaskListView/ControlButtons/UploadButton/__test__/UploadButton.test.js', () => {
    describe('UploadButton', () => {
        it('should display the button', () => {
            // Arrange
            const useStateMock = jest.spyOn(React, 'useState');
            useStateMock.mockReturnValueOnce([false, jest.fn()]);
            // Act
            const target = render(<UploadButton />);

            // Assert
            expect(target.getByRole("button")).toBeTruthy();
        });
    });
});