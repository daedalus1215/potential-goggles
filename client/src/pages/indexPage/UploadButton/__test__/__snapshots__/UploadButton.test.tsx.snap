// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/tasks/TaskListView/ControlButtons/UploadButton/__test__/UploadButton.test.js UploadButton should display the button 1`] = `
<button
  class="baseBtn IconButton hasTitle info"
  data-testid="icon-button"
>
  <i
    class="bi bi-cloud-upload icon"
  />
  <span
    class="title"
  >
    Upload
  </span>
</button>
`;
